File: backend/app/schemas/statistics.py
from datetime import date

from pydantic import BaseModel

from app.schemas.daily_report_day import DailyReportDayMonthItemData


class StatisticsMonthlyData(BaseModel):
    month: str
    effective_date_from: date
    effective_date_to: date
    denominator_days: int
    completed_days: int
    completion_rate: float | None
    daily_report_count: int
    weekly_report_count: int
    current_streak_days: int
    days: list[DailyReportDayMonthItemData]

File: electron/src/renderer/src/types/statistics.ts
import type { DailyReportDayMonthItemData } from './daily-report-day'

export interface StatisticsMonthlyData {
  month: string
  effective_date_from: string
  effective_date_to: string
  denominator_days: number
  completed_days: number
  completion_rate: number | null
  daily_report_count: number
  weekly_report_count: number
  current_streak_days: number
  days: DailyReportDayMonthItemData[]
}
